### PR TITLE
Cancelling label re-factor + Fix cancellation modal

### PR DESCRIPTION
**Pull request by biocom**
# Summary

- Re-factors the 'Cancelling' state:

https://user-images.githubusercontent.com/31534717/130212386-f51350e9-9904-4d52-b9b2-0e036fe8afee.mov

- Fixing an issue where on click of 'cancel' on the cancellation modal, it closes the modal+sidebar.

https://user-images.githubusercontent.com/31534717/130214224-73dca878-3297-453c-a3ff-ab41a208f112.mov

- Closes the orders side panel, on click of the FAQ link:
Screen Shot 2021-08-23 at 11 54 22

### CODE REVIEW

File: src/custom/components/OrdersPanel/index.tsx [hunk 1]
```diff
@@ -104,7 +104,7 @@ export interface OrdersPanelProps {
 export default function OrdersPanel({ closeOrdersPanel }: OrdersPanelProps) {
   // Close sidebar if clicked/tapped outside
   const ref = useRef<HTMLDivElement | null>(null)
-  useOnClickOutside(ref, closeOrdersPanel)
+  // useOnClickOutside(ref, closeOrdersPanel)
```
Comment (biocom) [inline src/custom/components/OrdersPanel/index.tsx:107]: WIP

**Review by elena-zh — APPROVED** [review]
Described changes LGTM!
Besides issues mentioned in https://github.com/gnosis/cowswap/issues/1279#pullrequestreview-733038179

**Review by alfetopito — APPROVED** [review]
Ah yes, much better!

Although, now we have 2 patterns for "pending" states.
1. Open: the left icon spins, the label is static
2. Cancelling: the left icon spins, the label shimmers

I personally think the shimmer -- although nice -- is unnecessary and could be removed to match Open orders.
But if we decide to keep it, it should be added to Open orders label too.

Thoughts?

**Comment by biocom** [comment]
> Ah yes, much better!
> 
> Although, now we have 2 patterns for "pending" states.
> 
> 1. Open: the left icon spins, the label is static
> 2. Cancelling: the left icon spins, the label shimmers
> 
> I personally think the shimmer -- although nice -- is unnecessary and could be removed to match Open orders.
> But if we decide to keep it, it should be added to Open orders label too.
> 
> Thoughts?

Yeah I get your thoughts. My initial thinking was, cancelling an order is something you actively are waiting for. A limit order might not have the same expectancy, but likely does (e.g. waiting for your order to be soon executed). I'm fine with removing it, as we already have the round spinner. Curious on thoughts of the others.

**Comment by elena-zh** [comment]
I would add shimmer labels to 'Open' order as well due to in a mobile version we do not have left animated icons at all.

**Comment by biocom** [comment]
@elena-zh @alfetopito Added the shimmer effect to the open orders as well. I like @elena-zh 's consideration that on mobile this is a welcome feature.

File: src/custom/components/AccountDetails/Transaction.tsx [hunk 2]
```diff
@@ -223,10 +220,10 @@ const StatusLabelWrapper = styled.div`
   justify-content: center;
 `
 
-const StatusLabel = styled.div<{ isPending: boolean }>`
+const StatusLabel = styled.div<{ isPending: boolean; isCancelling: boolean }>`
   height: 28px;
   width: 100px;
-  border: ${({ isPending, theme }) => isPending && `1px solid ${theme.border2}`};
+  border: ${({ isPending, isCancelling, theme }) => !isCancelling && isPending && `1px solid ${theme.border2}`};
```
Comment (W3stside) [inline src/custom/components/AccountDetails/Transaction.tsx:226]: i would make the whole property conditional not the value. else it's still rendering: `border: false;`

```suggestion
  ${({ isPending, isCancelling, theme }) => !isCancelling && isPending && `border:  1px solid ${theme.border2};`}
```

File: src/custom/components/OrdersPanel/index.tsx [hunk 3]
```diff
@@ -51,6 +49,20 @@ const SideBar = styled.div`
...
 `
 
+const SidebarBackground = styled.div`
+  position: fixed;
+  top: 0;
+  left: 0;
+  z-index: 0;
+  width: 100%;
+  height: 100%;
+  background: rgb(0 0 0 / 25%);
```
Comment (W3stside) [inline src/custom/components/OrdersPanel/index.tsx:59]: what is the / here? never seen

**Review by W3stside — APPROVED** [review]
very nice pr!

File: src/custom/components/OrdersPanel/index.tsx (hunk 3 above)
Comment (biocom) [reply src/custom/components/OrdersPanel/index.tsx:59]: RGB color with a percentage opacity (vs. RGBA (alpha setting opacity).

**Pull request merged**